app/controllers/TagManagementApi.scala

```diff
@@ -86,15 +86,38 @@ object TagManagementApi extends Controller with PanDomainAuthActions {
   def updateSection(id: Long) = APIAuthAction { req =>
 
     req.body.asJson.map { json =>
-      SectionRepository.updateSection(json).map{ section =>
-        Ok(Json.toJson(section))
-      }.getOrElse(BadRequest("Could not update section"))
+      try {
+        UpdateSectionCommand(json.as[Section]).process.map{ t => Ok(Json.toJson(t)) } getOrElse NotFound
+      } catch {
+        commandErrorAsResult
+      }
+    }.getOrElse {
+      BadRequest("Expecting Json data")
+    }
+  }
+
+  def addEditionToSection(id: Long) = APIAuthAction { req =>
+    req.body.asJson.map { json =>
+      val editionName = (json \ "editionName").as[String]
 
+      try {
```

Comment:
Might be easier to use the scala Try for this and the above ... should be a bit simpler to read.